master/internal/kubernetes/pod.go

```diff
@@ -80,6 +103,35 @@ func (p *pod) Receive(ctx *actor.Context) error {
 			return err
 		}
 
+	case podStatusUpdate:
+		if err := p.receivePodStatusUpdate(ctx, msg); err != nil {
+			return err
+		}
+
+	case sproto.ContainerLog:
+		p.receiveContainerLogs(ctx, msg)
+
+	case sproto.StopPod:
+		ctx.Log().WithField("pod", p.podName).Info("received request to stop pod")
+		if err := p.deleteKubernetesResources(ctx); err != nil {
+			return err
+		}
+
+	case actor.PostStop:
+		if p.logStreamer != nil {
+			p.logStreamer.Stop()
```

Comment:
non-blocking: I think the [actor system](https://github.com/determined-ai/determined/blob/068885322d544b0a8fb32e975f9d86e2cfe564bb/master/pkg/actor/ref.go#L262) should make sure the children of an actor are all stopped before the actor is stopped, which would mean this isn't necessary. Did you see something go wrong without this? If it works without this, then you don't need the `logStreamer` field at all.

Reply:
Yeah good call, this isn't necessary.